From 174885ce4f15067c2ff12bb0e0dc4c79b2fb64cc Mon Sep 17 00:00:00 2001
From: Cheick Traore <cheick.traore-ext@st.com>
Date: Tue, 27 Aug 2024 18:35:31 +0200
Subject: [PATCH 1106/1695] dts: st: add ADC nodes on stm32mp211

Add ADC1 and ADC2 nodes on STM32MP21 SoC. Also define internal voltage
channels for each ADCs, exception made for vbat/4 channel (to avoid
undesired load on it, in case no conversion is needed on this channel).
Choice is let to board implementation (via DT) to define it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 214114e7b301f56e929a93f70c3c997eef695f41 dts: st: add ADC nodes on stm32mp211]
Change-Id: I9fd5abe9366b94715c8ac84f999f9a881066e0b3
Signed-off-by: Cheick Traore <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/399603
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 68 +++++++++++++++++++++++++++++++
 1 file changed, 68 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 33efc19fb..b58d24d97 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -375,6 +375,74 @@
 				status = "disabled";
 			};
 
+			adc_1: adc@404e0000 {
+				compatible = "st,stm32mp21-adc-core";
+				reg = <0x404e0000 0x400>;
+				interrupts = <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_ADC1>;
+				clock-names = "adc";
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP21_RIFSC_ADC1_ID>;
+				status = "disabled";
+
+				adc1: adc@0 {
+					compatible = "st,stm32mp21-adc";
+					#io-channel-cells = <1>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0x0>;
+					interrupt-parent = <&adc_1>;
+					interrupts = <0>;
+					status = "disabled";
+
+					channel@14 {
+						reg = <14>;
+						label = "vrefint";
+					};
+				};
+			};
+
+			adc_2: adc@404f0000 {
+				compatible = "st,stm32mp21-adc-core";
+				reg = <0x404f0000 0x400>;
+				interrupts = <GIC_SPI 82 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_ADC2>;
+				clock-names = "adc";
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP21_RIFSC_ADC2_ID>;
+				status = "disabled";
+
+				adc2: adc@0 {
+					compatible = "st,stm32mp21-adc";
+					#io-channel-cells = <1>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0x0>;
+					interrupt-parent = <&adc_2>;
+					interrupts = <0>;
+					status = "disabled";
+
+					channel@14 {
+						reg = <14>;
+						label = "vrefint";
+					};
+					channel@15 {
+						reg = <15>;
+						label = "vddcore";
+					};
+					channel@17 {
+						reg = <17>;
+						label = "vddcpu";
+					};
+				};
+			};
+
 			hash2: hash@42010000  {
 				compatible = "st,stm32mp13-hash";
 				reg = <0x42010000  0x400>;
-- 
2.47.2

